### DISCUSSION TITLE
How to import AWS AppSync types and directives

### DISCUSSION THREAD

**Discussion by RadoslavMarinov** [graphql-hive/graphql-eslint#1063] [General]
I am using AWS CDK to build our AppSync project. I have defined a schema file `scheam.graphql` where we utilize AWS [scapar types](https://docs.aws.amazon.com/appsync/latest/devguide/scalars.html) and directives such as `@aws_api_key` and `@aws_subscribe`.

The problem is that the graphql-eslint does not recognize these types and directives and reports errors.
Any  idea how (or from where) to import the AWS AppSync schema with these directives?

**Comment by dimaMachina** [comment]
Hi @RadoslavMarinov,
I don't know where you can import these scalars and directives but you can extend your schema manually in eslintrc config `parserOptions.schema`
```js
      parserOptions: {
        schema: [
          'schema.graphql',
          /* GraphQL */ `
            scalar AWSDateTime
            # ...
          `,
        ],
      },
```

**Reply by RadoslavMarinov** [reply]
For some reason this breaks my 

```
'@graphql-eslint/known-directives': [
          'error',
          { ignoreClientDirectives: ['aws_api_key'] }
        ]
```
rule and now Im getting ` Unknown directive "@aws_api_key".`

**Reply by RadoslavMarinov** [reply]
Removed rule and defined the directive into the schema:

```
schema: [
          './schema.graphql',
          /* GraphQL */
          `
            scalar AWSJSON
            directive @aws_api_key(
              reason: String = "No longer supported"
            ) on FIELD_DEFINITION | ENUM_VALUE | OBJECT
          `
        ]
```

Now it works ;)
